lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -20,17 +20,15 @@ def sync
         if st.success?
           pods_json = JSON.parse(pod_list)["items"]
           pods_json.each do |pod_json|
-            pod_name = pod_json["metadata"]["name"]
             pod = Pod.new(
-              name: pod_name,
               namespace: namespace,
               context: context,
-              file: nil,
+              definition: pod_json,
+              logger: @logger,
               parent: "#{@name.capitalize} deployment",
-              logger: @logger
+              deploy_started: @deploy_started
             )
-            pod.deploy_started = @deploy_started
-            pod.interpret_json_data(pod_json)
+            pod.sync(pod_json)
```

Comment:
Just an idea for future refactorings: if `pod.sync` doesn't actually sync and just assigns the definition from `pod_json`, could this be as simple as:

```ruby
pod.definition = pod_json
```

(with some setter logic like `log_suspicious_states`)